src/webgl/shaders/imageLight.frag

```diff
@@ -0,0 +1,113 @@
+precision mediump float;
+varying vec3 localPos;
+
+// the HDR cubemap converted (can be from an equirectangular environment map.)
+uniform sampler2D environmentMap;
+varying vec2 vTexCoord;
+
+const float PI = 3.141;
+
+vec2 normalToEquirectangular( vec3 v){
+    vec2 uv;
+    // taking the arctangent of the v.z and v.x components and dividing it by the circumference of a circle
+    // Adding 0.5 ensures that the resulting value is in the range [0, 1].
+    uv.x = atan(v.z, v.x) / 3.14159 + 0.5;
+    // taking the arcsine of the v.y component and dividing it by the half circumference of a circle
+    uv.y = asin(v.y) / 3.14159 + 0.5;
+  
+    
+    return uv;
+}
+
+vec2 nTOE( vec3 v ){
+  // x = r sin(phi) cos(theta)   
+  // y = r cos(phi)  
+  // z = r sin(phi) sin(theta)
+  float phi = acos( v.y );
+  // if phi is 0, then there are no x, z components
+  float theta = 0.0;
+  // else 
+  theta = acos(v.x / sin(phi));
+  float sinTheta = v.z / sin(phi);
+  if (sinTheta < 0.0) {
+    // Turn it into -theta, but in the 0-2PI range
+    theta = 2.0 * PI - theta;
+  }
+  theta = theta / (2.0 * 3.14159);
```

Comment:
Maybe where we use 3.14159 right now we can use the `const float PI` we defined earlier, and add more decimals to that?

Reply:
Sure, I'll clean up this file. It has some extra code as well.